src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1586,6 +1586,7 @@ static void WindowRideConstructionConstruct(rct_window* w)
 static void WindowRideConstructionMouseupDemolish(rct_window* w)
 {
     int32_t direction;
+    Direction currentDirection = _currentTrackPieceDirection;
```

Comment:
Let’s add a comment here explain why we need to save the direction (because it is reset by `ride_initialise_construction_window()`).

Reply:
Good idea, thanks.